cdap-watchdog/src/main/java/co/cask/cdap/metrics/collect/AggregatedMetricsCollectionService.java

```diff
@@ -86,7 +91,33 @@ protected final void runOneIteration() throws Exception {
     Iterator<MetricValue> metricsItor = getMetrics(timestamp);
 
     try {
-      publish(metricsItor);
+      publish(metricsItor, new MetaMetricsComputer() {
+        private List<MetricValue> metaMetrics = Lists.newArrayList();
+        private int numMetrics = 0;
+
+        @Override
+        public void visitMetric(MetricValue metricValue) {
+          numMetrics++;
+
+          long timestampMs = TimeUnit.MILLISECONDS.convert(metricValue.getTimestamp(), TimeUnit.SECONDS);
```

Comment:
I think using `TimeUnit.SECONDS.toMillis(metricValue.getTimestamp());` is simpler